src/main/java/io/airlift/compress/lz4/Lz4Compressor.java

```diff
@@ -38,18 +40,20 @@ public int maxCompressedLength(int uncompressedSize)
     @Override
     public int compress(byte[] input, int inputOffset, int inputLength, byte[] output, int outputOffset, int maxOutputLength)
     {
-        Arrays.fill(table, 0);
+        int tableSize = computeTableSize(inputLength);
+        Arrays.fill(table, 0, tableSize, 0);
 
         long inputAddress = ARRAY_BYTE_BASE_OFFSET + inputOffset;
         long outputAddress = ARRAY_BYTE_BASE_OFFSET + outputOffset;
 
-        return Lz4RawCompressor.compress(input, inputAddress, inputLength, output, outputAddress, maxOutputLength, table);
+        return Lz4RawCompressor.compress(input, inputAddress, inputLength, output, outputAddress, maxOutputLength, table, tableSize);
```

Comment:
I find it a bit strange that you calculate and initialize the table outside of the compressor instead of leaving these private details to the raw compressor code.  It is also strange that there is a min table size but it is defined outside the compressor logic